+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+                            z_ijk = obj.w.z(ii,jj,opts.n_s+1);
+                            lambda_ijk = obj.w.lambda(ii,jj,opts.n_s+1);
+                            mu_not_c_ij = obj.w.mu_not_c(ii,jj);
+                            not_c_primals_ij = obj.w.not_c_primals(ii,jj);
+                            f_ijk = dcs.f_fun(x_ijk, z_ijk, ui, v_global, p);
+                            nabla_c_ijk = dcs.nabla_c_fun(x_ijk, z_ijk, v_global, p);
+                            lambda_cf_ijk = inv(nabla_c_ijk'*nabla_c_ijk)*nabla_c_ijk'*f_ijk;
+
+                            c = not_c_primals_ij(1); d = not_c_primals_ij(2);
+                            c_ijk = dcs.c_fun(x_ijk, z_ijk, v_global, p);
+                            obj.g.not_c(ii,jj) = {[c + d*c_ijk - 1;1-mu_not_c_ij(1) - mu_not_c_ij(2);mu_not_c_ij(1)*c_ijk + mu_not_c_ij(3)], [0;0;0], [inf;0;0]};
+                            obj.G.not_c(ii,jj) = {[c + d*c_ijk - 1;c;d]};
+                            obj.H.not_c(ii,jj) = {[mu_not_c_ij]};
+                            obj.g.lambda_explicit(ii,jj) = {[lambda_ijk-c*lambda_cf_ijk]};
+                            obj.g.dynamics(ii,jj,opts.n_s+1) = {x_ijk - x_ij_end};
+                            obj.g.c_lb(ii,jj,opts.n_s+1) = {dcs.c_fun(x_ijk, z_ijk, v_global, p), 0, inf};
+                            obj.g.z(ii,jj,opts.n_s+1) = {dcs.g_z_fun(x_ijk, z_ijk, ui, v_global, p)};
+                        end
+                        if ~opts.g_path_at_stg && opts.g_path_at_fe
+                            obj.g.path(ii,jj) = {dcs.g_path_fun(x_ijk, z_ijk, ui, v_global, p), model.lbg_path, model.ubg_path};
+                        end
+                      case RKRepresentation.differential
+                        error("Differential representation without lifting is unsupported for gradient complementarity systems")
+                        
+                      case RKRepresentation.differential_lift_x
+                        % In differential representation with lifted state stage variables are the state derivatives and we
+                        % lift the states at each stage point as well.
+                        for kk = 1:opts.n_s
+                            x_ijk = obj.w.x(ii,jj,kk);
+                            x_temp = x_prev;
+                            for rr = 1:opts.n_s
+                                x_temp = x_temp + h*opts.A_rk(kk,rr)*obj.w.v(ii,jj,rr);
+                            end
+                            obj.g.lift_x(ii,jj,kk) = {x_ijk - x_temp};
+                        end
+                        x_ij_end = x_prev;
+                        for kk=1:opts.n_s
+                            x_ijk = obj.w.x(ii,jj,kk);
+                            v_ijk = obj.w.v(ii,jj,kk);
+                            z_ijk = obj.w.z(ii,jj,kk);
+                            lambda_ijk = obj.w.lambda(ii,jj,kk);
+                           
+                            fj = s_sot*dcs.f_x_fun(x_ijk, z_ijk, lambda_ijk, ui, v_global, p);
```

Comment:
for sake of consistency, f_j, q_j ? (and elsewhere)

Reply:
`f_ijk` and `q_ijk` is better, agreed.

Reply:
I assume we have the same in Stewart and Heaviside 🥹

Reply:
Also fixed :smile: